java/kotlin-extractor2/src/main/kotlin/entities/TypeParameter.kt

```diff
@@ -0,0 +1,85 @@
+package com.github.codeql.entities
+
+import com.github.codeql.*
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.analysis.api.KaSession
+import org.jetbrains.kotlin.analysis.api.symbols.KaTypeParameterSymbol
+import org.jetbrains.kotlin.analysis.api.symbols.psiSafe
+import org.jetbrains.kotlin.analysis.api.types.KaTypeParameterType
+import org.jetbrains.kotlin.types.Variance
+
+context(KaSession)
+fun KotlinUsesExtractor.getTypeParameterLabel(param: KaTypeParameterSymbol): String {
+    // Use this instead of `useDeclarationParent` so we can use useFunction with noReplace = true,
+    // ensuring that e.g. a method-scoped type variable declared on kotlin.String.transform<R> gets
+    // a different name to the corresponding java.lang.String.transform<R>, even though
+    // useFunction will usually replace references to one function with the other.
+    val parentLabel = getTypeParameterParentLabel(param)
```

Comment:
I think this wants to fail, or possibly return some other best-attempt string, if `parentLabel` is null